packages/wrangler/src/dev/dev.tsx

```diff
@@ -274,17 +278,88 @@ type DevSessionProps = DevProps & {
 };
 
 function DevSession(props: DevSessionProps) {
+	const [accountId, setAccountIdStateOnly] = useState(props.accountId);
+	const accountIdDeferred = useMemo(() => createDeferred<string>(), []);
+	const setAccountIdAndResolveDeferred = useCallback(
+		(newAccountId: string) => {
+			setAccountIdStateOnly(newAccountId);
+			accountIdDeferred.resolve(newAccountId);
+		},
+		[setAccountIdStateOnly, accountIdDeferred]
+	);
+
+	useEffect(() => {
+		if (props.accountId) {
+			setAccountIdAndResolveDeferred(props.accountId);
+		}
+
+		// run once on mount only (to synchronize the deferred value with the pre-selected props.accountId)
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
 	const [devEnv] = useState(() => new DevEnv());
 	useEffect(() => {
 		return () => {
 			void devEnv.teardown();
 		};
 	}, [devEnv]);
-	const startDevWorkerOptions: StartDevWorkerOptions = useMemo(
-		() => ({
+	const startDevWorkerOptions: StartDevWorkerOptions = useMemo(() => {
+		const routes =
+			props.routes?.map((r) =>
+				typeof r === "string"
+					? ({
+							type: "route",
+							pattern: r,
+						} as const)
+					: ({ type: "route", ...r } as const)
```

Comment:
Another option would be to define `Trigger` as the generic argument to `map` in these three definitions. That could provide more type safety if we come to modify or expand these object constructions.

Reply:
That's fair. It would need to be `Extract<Trigger, { type: '...' }>` for each .map